include/deal.II/base/hdf5.h

```diff
@@ -0,0 +1,1073 @@
+     * H5D_MPIO_NOT_SIMPLE_OR_SCALAR_DATASPACES   | Collective I/O was not performed because one of the dataspaces was neither simple nor scalar.
+     * H5D_MPIO_POINT_SELECTIONS                  | Collective I/O was not performed because there were point selections in one of the dataspaces.
+     * H5D_MPIO_NOT_CONTIGUOUS_OR_CHUNKED_DATASET | Collective I/O was not performed because the dataset was neither contiguous nor chunked.
+     * H5D_MPIO_FILTERS                           | Collective I/O was not performed because filters needed to be applied.
+     */
+    std::string
+    get_global_no_collective_cause();
+
+    /**
+     * This function returns the global causes that broke collective I/O on the
+     * last parallel I/O call. See <a
+     * href="https://support.hdfgroup.org/HDF5/doc/RM/RM_H5P.html#Property-GetMpioNoCollectiveCause">H5Pget_mpio_no_collective_cause</a>.
+     * The return type is `uint32_t` and corresponds to the value returned by
+     * H5Pget_mpio_no_collective_cause.
+     *
+     * The return value value can be
+     * Value                                      | Meaning
+     * ------------------------------------------ | -------
+     * H5D_MPIO_COLLECTIVE                        | Collective I/O was performed successfully.
+     * H5D_MPIO_SET_INDEPENDENT                   | Collective I/O was not performed because independent I/O was requested.
+     * H5D_MPIO_DATATYPE_CONVERSION               | Collective I/O was not performed because datatype conversions were required.
+     * H5D_MPIO_DATA_TRANSFORMS                   | Collective I/O was not performed because data transforms needed to be applied.
+     * H5D_MPIO_SET_MPIPOSIX                      | Collective I/O was not performed because the selected file driver was MPI-POSIX.
+     * H5D_MPIO_NOT_SIMPLE_OR_SCALAR_DATASPACES   | Collective I/O was not performed because one of the dataspaces was neither simple nor scalar.
+     * H5D_MPIO_POINT_SELECTIONS                  | Collective I/O was not performed because there were point selections in one of the dataspaces.
+     * H5D_MPIO_NOT_CONTIGUOUS_OR_CHUNKED_DATASET | Collective I/O was not performed because the dataset was neither contiguous nor chunked.
+     * H5D_MPIO_FILTERS                           | Collective I/O was not performed because filters needed to be applied.
+     */
+    uint32_t
+    get_global_no_collective_cause_as_hdf5_type();
+
+    /**
+     * This function returns the boolean query_io_mode.
+     *
+     * In cases where maximum performance has to be achieved, it is important to
+     * make sure that all MPI read/write operations are collective. The HDF5
+     * library provides API routines that can be used after the read/write I/O
+     * operations to query the I/O mode. If query_io_mode is set to true, then
+     * after every read/write operation the HDF5 deal.ii interface calls the
```

Comment:
deal.II

Reply:
actually, I think it should be "deal.II's HDF5 interface" to make this completely unambiguous who wraps whom.